c2corg_api/tests/views/test_book.py

```diff
@@ -0,0 +1,571 @@
+                'activities': ['hiking'],
+                'book_types': ['biography'],
+                'locales': [
+                    {'lang': 'en', 'title': 'Escalades au Thaurac',
+                     'version': self.locale_en.version}
+                ]
+            }
+        }
+        self.put_wrong_document_id(body, user='moderator')
+
+    def test_put_wrong_document_version(self):
+        body = {
+            'document': {
+                'document_id': self.book1.document_id,
+                'version': -9999,
+                'activities': ['hiking'],
+                'book_types': ['biography'],
+                'locales': [
+                    {'lang': 'en', 'title': 'Escalades au Thaurac',
+                     'version': self.locale_en.version}
+                ]
+            }
+        }
+        self.put_wrong_version(
+            body, self.book1.document_id, user='moderator')
+
+    def test_put_wrong_locale_version(self):
+        body = {
+            'document': {
+                'document_id': self.book1.document_id,
+                'version': self.book1.version,
+                'activities': ['hiking'],
+                'book_types': ['biography'],
+                'locales': [
+                    {'lang': 'en', 'title': 'Escalades au Thaurac',
+                     'version': -9999}
+                ]
+            }
+        }
+        self.put_wrong_version(
+            body, self.book1.document_id, user='moderator')
+
+    def test_put_wrong_ids(self):
+        body = {
+            'document': {
+                'document_id': self.book1.document_id,
+                'version': self.book1.version,
+                'activities': ['hiking'],
+                'book_types': ['biography'],
+                'locales': [
+                    {'lang': 'en', 'title': 'Escalades au Thaurac',
+                     'version': self.locale_en.version}
+                ]
+            }
+        }
+        self.put_wrong_ids(body, self.book1.document_id, user='moderator')
+
+    def test_put_no_document(self):
+        self.put_put_no_document(self.book1.document_id, user='moderator')
+
+    def test_put_success_all(self):
+        body = {
+            'message': 'Update',
+            'document': {
+                'document_id': self.book1.document_id,
+                'version': self.book1.version,
+                'quality': quality_types[1],
+                'activities': ['hiking'],
+                'book_types': ['biography'],
+                'associations': {
+                    'articles': [
+                        {'document_id': self.article2.document_id}
+                    ],
+                    'images': [
+                        {'document_id': self.image2.document_id}
+                    ],
+                    'routes': [
+                        {'document_id': self.route2.document_id}
+                    ]
+                }
+                # ,
```

Comment:
Remove this block if it is not needed. Or did you explicitly want to test that no geometry can be added?